File: windows/Liferay.SDK/Service/V73/Commerceshipment/CommerceshipmentService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Commerceshipment
{
	public class CommerceshipmentService : ServiceBase
	{
		public CommerceshipmentService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> UpdateAddressAsync(long commerceShipmentId, string name, string description, string street1, string street2, string street3, string city, string zip, long commerceRegionId, long commerceCountryId, string phoneNumber)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceShipmentId", commerceShipmentId);
			_parameters.Add("name", name);
			_parameters.Add("description", description);
			_parameters.Add("street1", street1);
			_parameters.Add("street2", street2);
			_parameters.Add("street3", street3);
			_parameters.Add("city", city);
			_parameters.Add("zip", zip);
			_parameters.Add("commerceRegionId", commerceRegionId);
			_parameters.Add("commerceCountryId", commerceCountryId);
			_parameters.Add("phoneNumber", phoneNumber);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/update-address", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateStatusAsync(long commerceShipmentId, int status)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceShipmentId", commerceShipmentId);
			_parameters.Add("status", status);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/update-status", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task DeleteCommerceShipmentAsync(long commerceShipmentId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceShipmentId", commerceShipmentId);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/delete-commerce-shipment", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task DeleteCommerceShipmentAsync(long commerceShipmentId, bool restoreStockQuantity)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceShipmentId", commerceShipmentId);
			_parameters.Add("restoreStockQuantity", restoreStockQuantity);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/delete-commerce-shipment", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task<dynamic> AddCommerceShipmentAsync(long groupId, long commerceAccountId, long commerceAddressId, long commerceShippingMethodId, string commerceShippingOptionName, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("commerceAccountId", commerceAccountId);
			_parameters.Add("commerceAddressId", commerceAddressId);
			_parameters.Add("commerceShippingMethodId", commerceShippingMethodId);
			_parameters.Add("commerceShippingOptionName", commerceShippingOptionName);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/add-commerce-shipment", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> AddCommerceShipmentAsync(long commerceOrderId, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceOrderId", commerceOrderId);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/add-commerce-shipment", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> GetCommerceShipmentAsync(long commerceShipmentId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceShipmentId", commerceShipmentId);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/get-commerce-shipment", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetCommerceShipmentsAsync(long companyId, long commerceAddressId, int start, int end, JsonObjectWrapper orderByComparator)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("commerceAddressId", commerceAddressId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "orderByComparator", "$languageUtil.getJSONWrapperClassName($parameter.type)", orderByComparator);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/get-commerce-shipments", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetCommerceShipmentsAsync(long companyId, IEnumerable<long> groupIds, IEnumerable<long> commerceAccountIds, string keywords, IEnumerable<int> shipmentStatuses, bool excludeShipmentStatus, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("groupIds", groupIds);
			_parameters.Add("commerceAccountIds", commerceAccountIds);
			_parameters.Add("keywords", keywords);
			_parameters.Add("shipmentStatuses", shipmentStatuses);
			_parameters.Add("excludeShipmentStatus", excludeShipmentStatus);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/get-commerce-shipments", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetCommerceShipmentsAsync(long companyId, int start, int end, JsonObjectWrapper orderByComparator)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "orderByComparator", "$languageUtil.getJSONWrapperClassName($parameter.type)", orderByComparator);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/get-commerce-shipments", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetCommerceShipmentsAsync(long companyId, int status, int start, int end, JsonObjectWrapper orderByComparator)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("status", status);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "orderByComparator", "$languageUtil.getJSONWrapperClassName($parameter.type)", orderByComparator);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/get-commerce-shipments", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetCommerceShipmentsCountAsync(long companyId, long commerceAddressId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("commerceAddressId", commerceAddressId);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/get-commerce-shipments-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<long> GetCommerceShipmentsCountAsync(long companyId, IEnumerable<long> groupIds, IEnumerable<long> commerceAccountIds, string keywords, IEnumerable<int> shipmentStatuses, bool excludeShipmentStatus)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("groupIds", groupIds);
			_parameters.Add("commerceAccountIds", commerceAccountIds);
			_parameters.Add("keywords", keywords);
			_parameters.Add("shipmentStatuses", shipmentStatuses);
			_parameters.Add("excludeShipmentStatus", excludeShipmentStatus);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/get-commerce-shipments-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<long> GetCommerceShipmentsCountAsync(long companyId, int status)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("status", status);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/get-commerce-shipments-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<long> GetCommerceShipmentsCountAsync(long companyId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/get-commerce-shipments-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<dynamic> UpdateCarrierDetailsAsync(long commerceShipmentId, string carrier, string trackingNumber)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceShipmentId", commerceShipmentId);
			_parameters.Add("carrier", carrier);
			_parameters.Add("trackingNumber", trackingNumber);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/update-carrier-details", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateCommerceShipmentAsync(long commerceShipmentId, string carrier, string trackingNumber, int status, int shippingDateMonth, int shippingDateDay, int shippingDateYear, int shippingDateHour, int shippingDateMinute, int expectedDateMonth, int expectedDateDay, int expectedDateYear, int expectedDateHour, int expectedDateMinute)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceShipmentId", commerceShipmentId);
			_parameters.Add("carrier", carrier);
			_parameters.Add("trackingNumber", trackingNumber);
			_parameters.Add("status", status);
			_parameters.Add("shippingDateMonth", shippingDateMonth);
			_parameters.Add("shippingDateDay", shippingDateDay);
			_parameters.Add("shippingDateYear", shippingDateYear);
			_parameters.Add("shippingDateHour", shippingDateHour);
			_parameters.Add("shippingDateMinute", shippingDateMinute);
			_parameters.Add("expectedDateMonth", expectedDateMonth);
			_parameters.Add("expectedDateDay", expectedDateDay);
			_parameters.Add("expectedDateYear", expectedDateYear);
			_parameters.Add("expectedDateHour", expectedDateHour);
			_parameters.Add("expectedDateMinute", expectedDateMinute);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/update-commerce-shipment", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateCommerceShipmentAsync(long commerceShipmentId, string name, string description, string street1, string street2, string street3, string city, string zip, long commerceRegionId, long commerceCountryId, string phoneNumber, string carrier, string trackingNumber, int status, int shippingDateMonth, int shippingDateDay, int shippingDateYear, int shippingDateHour, int shippingDateMinute, int expectedDateMonth, int expectedDateDay, int expectedDateYear, int expectedDateHour, int expectedDateMinute)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceShipmentId", commerceShipmentId);
			_parameters.Add("name", name);
			_parameters.Add("description", description);
			_parameters.Add("street1", street1);
			_parameters.Add("street2", street2);
			_parameters.Add("street3", street3);
			_parameters.Add("city", city);
			_parameters.Add("zip", zip);
			_parameters.Add("commerceRegionId", commerceRegionId);
			_parameters.Add("commerceCountryId", commerceCountryId);
			_parameters.Add("phoneNumber", phoneNumber);
			_parameters.Add("carrier", carrier);
			_parameters.Add("trackingNumber", trackingNumber);
			_parameters.Add("status", status);
			_parameters.Add("shippingDateMonth", shippingDateMonth);
			_parameters.Add("shippingDateDay", shippingDateDay);
			_parameters.Add("shippingDateYear", shippingDateYear);
			_parameters.Add("shippingDateHour", shippingDateHour);
			_parameters.Add("shippingDateMinute", shippingDateMinute);
			_parameters.Add("expectedDateMonth", expectedDateMonth);
			_parameters.Add("expectedDateDay", expectedDateDay);
			_parameters.Add("expectedDateYear", expectedDateYear);
			_parameters.Add("expectedDateHour", expectedDateHour);
			_parameters.Add("expectedDateMinute", expectedDateMinute);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/update-commerce-shipment", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateExpectedDateAsync(long commerceShipmentId, int expectedDateMonth, int expectedDateDay, int expectedDateYear, int expectedDateHour, int expectedDateMinute)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceShipmentId", commerceShipmentId);
			_parameters.Add("expectedDateMonth", expectedDateMonth);
			_parameters.Add("expectedDateDay", expectedDateDay);
			_parameters.Add("expectedDateYear", expectedDateYear);
			_parameters.Add("expectedDateHour", expectedDateHour);
			_parameters.Add("expectedDateMinute", expectedDateMinute);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/update-expected-date", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateShippingDateAsync(long commerceShipmentId, int shippingDateMonth, int shippingDateDay, int shippingDateYear, int shippingDateHour, int shippingDateMinute)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceShipmentId", commerceShipmentId);
			_parameters.Add("shippingDateMonth", shippingDateMonth);
			_parameters.Add("shippingDateDay", shippingDateDay);
			_parameters.Add("shippingDateYear", shippingDateYear);
			_parameters.Add("shippingDateHour", shippingDateHour);
			_parameters.Add("shippingDateMinute", shippingDateMinute);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/update-shipping-date", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetCommerceShipmentsByOrderIdAsync(long commerceOrderId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceOrderId", commerceOrderId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/get-commerce-shipments-by-order-id", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetCommerceShipmentsCountByOrderIdAsync(long commerceOrderId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceOrderId", commerceOrderId);

			var _command = new JsonObject()
			{
				{ "/commerce.commerceshipment/get-commerce-shipments-count-by-order-id", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}
	}
}

File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/commerceshipment/CommerceshipmentService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.commerceshipment;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.JSONObjectWrapper;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class CommerceshipmentService extends BaseService {

	public CommerceshipmentService(Session session) {
		super(session);
	}

	public JSONObject addCommerceShipment(long commerceOrderId, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceOrderId", commerceOrderId);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/commerce.commerceshipment/add-commerce-shipment", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject addCommerceShipment(long groupId, long commerceAccountId, long commerceAddressId, long commerceShippingMethodId, String commerceShippingOptionName, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceAccountId", commerceAccountId);
			_params.put("commerceAddressId", commerceAddressId);
			_params.put("commerceShippingMethodId", commerceShippingMethodId);
			_params.put("commerceShippingOptionName", checkNull(commerceShippingOptionName));
			_params.put("groupId", groupId);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/commerce.commerceshipment/add-commerce-shipment", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public void deleteCommerceShipment(long commerceShipmentId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceShipmentId", commerceShipmentId);

			_command.put("/commerce.commerceshipment/delete-commerce-shipment", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

	public void deleteCommerceShipment(long commerceShipmentId, boolean restoreStockQuantity) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceShipmentId", commerceShipmentId);
			_params.put("restoreStockQuantity", restoreStockQuantity);

			_command.put("/commerce.commerceshipment/delete-commerce-shipment", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

	public JSONObject getCommerceShipment(long commerceShipmentId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceShipmentId", commerceShipmentId);

			_command.put("/commerce.commerceshipment/get-commerce-shipment", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONArray getCommerceShipments(long companyId, int status, int start, int end, JSONObjectWrapper orderByComparator) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);
			_params.put("end", end);
			_params.put("start", start);
			_params.put("status", status);
			mangleWrapper(_params, "orderByComparator", "com.liferay.portal.kernel.util.OrderByComparator<com.liferay.commerce.model.CommerceShipment>", orderByComparator);

			_command.put("/commerce.commerceshipment/get-commerce-shipments", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public JSONArray getCommerceShipments(long companyId, int start, int end, JSONObjectWrapper orderByComparator) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);
			_params.put("end", end);
			_params.put("start", start);
			mangleWrapper(_params, "orderByComparator", "com.liferay.portal.kernel.util.OrderByComparator<com.liferay.commerce.model.CommerceShipment>", orderByComparator);

			_command.put("/commerce.commerceshipment/get-commerce-shipments", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public JSONArray getCommerceShipments(long companyId, JSONArray groupIds, JSONArray commerceAccountIds, String keywords, JSONArray shipmentStatuses, boolean excludeShipmentStatus, int start, int end) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceAccountIds", checkNull(commerceAccountIds));
			_params.put("companyId", companyId);
			_params.put("end", end);
			_params.put("excludeShipmentStatus", excludeShipmentStatus);
			_params.put("groupIds", checkNull(groupIds));
			_params.put("keywords", checkNull(keywords));
			_params.put("shipmentStatuses", checkNull(shipmentStatuses));
			_params.put("start", start);

			_command.put("/commerce.commerceshipment/get-commerce-shipments", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public JSONArray getCommerceShipments(long companyId, long commerceAddressId, int start, int end, JSONObjectWrapper orderByComparator) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceAddressId", commerceAddressId);
			_params.put("companyId", companyId);
			_params.put("end", end);
			_params.put("start", start);
			mangleWrapper(_params, "orderByComparator", "com.liferay.portal.kernel.util.OrderByComparator<com.liferay.commerce.model.CommerceShipment>", orderByComparator);

			_command.put("/commerce.commerceshipment/get-commerce-shipments", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public JSONArray getCommerceShipmentsByOrderId(long commerceOrderId, int start, int end) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceOrderId", commerceOrderId);
			_params.put("end", end);
			_params.put("start", start);

			_command.put("/commerce.commerceshipment/get-commerce-shipments-by-order-id", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public Integer getCommerceShipmentsCount(long companyId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);

			_command.put("/commerce.commerceshipment/get-commerce-shipments-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public Integer getCommerceShipmentsCount(long companyId, int status) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);
			_params.put("status", status);

			_command.put("/commerce.commerceshipment/get-commerce-shipments-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public Integer getCommerceShipmentsCount(long companyId, JSONArray groupIds, JSONArray commerceAccountIds, String keywords, JSONArray shipmentStatuses, boolean excludeShipmentStatus) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceAccountIds", checkNull(commerceAccountIds));
			_params.put("companyId", companyId);
			_params.put("excludeShipmentStatus", excludeShipmentStatus);
			_params.put("groupIds", checkNull(groupIds));
			_params.put("keywords", checkNull(keywords));
			_params.put("shipmentStatuses", checkNull(shipmentStatuses));

			_command.put("/commerce.commerceshipment/get-commerce-shipments-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public Integer getCommerceShipmentsCount(long companyId, long commerceAddressId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceAddressId", commerceAddressId);
			_params.put("companyId", companyId);

			_command.put("/commerce.commerceshipment/get-commerce-shipments-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public Integer getCommerceShipmentsCountByOrderId(long commerceOrderId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceOrderId", commerceOrderId);

			_command.put("/commerce.commerceshipment/get-commerce-shipments-count-by-order-id", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public JSONObject updateAddress(long commerceShipmentId, String name, String description, String street1, String street2, String street3, String city, String zip, long commerceRegionId, long commerceCountryId, String phoneNumber) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("city", checkNull(city));
			_params.put("commerceCountryId", commerceCountryId);
			_params.put("commerceRegionId", commerceRegionId);
			_params.put("commerceShipmentId", commerceShipmentId);
			_params.put("description", checkNull(description));
			_params.put("name", checkNull(name));
			_params.put("phoneNumber", checkNull(phoneNumber));
			_params.put("street1", checkNull(street1));
			_params.put("street2", checkNull(street2));
			_params.put("street3", checkNull(street3));
			_params.put("zip", checkNull(zip));

			_command.put("/commerce.commerceshipment/update-address", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateCarrierDetails(long commerceShipmentId, String carrier, String trackingNumber) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("carrier", checkNull(carrier));
			_params.put("commerceShipmentId", commerceShipmentId);
			_params.put("trackingNumber", checkNull(trackingNumber));

			_command.put("/commerce.commerceshipment/update-carrier-details", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateCommerceShipment(long commerceShipmentId, String carrier, String trackingNumber, int status, int shippingDateMonth, int shippingDateDay, int shippingDateYear, int shippingDateHour, int shippingDateMinute, int expectedDateMonth, int expectedDateDay, int expectedDateYear, int expectedDateHour, int expectedDateMinute) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("carrier", checkNull(carrier));
			_params.put("commerceShipmentId", commerceShipmentId);
			_params.put("expectedDateDay", expectedDateDay);
			_params.put("expectedDateHour", expectedDateHour);
			_params.put("expectedDateMinute", expectedDateMinute);
			_params.put("expectedDateMonth", expectedDateMonth);
			_params.put("expectedDateYear", expectedDateYear);
			_params.put("shippingDateDay", shippingDateDay);
			_params.put("shippingDateHour", shippingDateHour);
			_params.put("shippingDateMinute", shippingDateMinute);
			_params.put("shippingDateMonth", shippingDateMonth);
			_params.put("shippingDateYear", shippingDateYear);
			_params.put("status", status);
			_params.put("trackingNumber", checkNull(trackingNumber));

			_command.put("/commerce.commerceshipment/update-commerce-shipment", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateCommerceShipment(long commerceShipmentId, String name, String description, String street1, String street2, String street3, String city, String zip, long commerceRegionId, long commerceCountryId, String phoneNumber, String carrier, String trackingNumber, int status, int shippingDateMonth, int shippingDateDay, int shippingDateYear, int shippingDateHour, int shippingDateMinute, int expectedDateMonth, int expectedDateDay, int expectedDateYear, int expectedDateHour, int expectedDateMinute) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("carrier", checkNull(carrier));
			_params.put("city", checkNull(city));
			_params.put("commerceCountryId", commerceCountryId);
			_params.put("commerceRegionId", commerceRegionId);
			_params.put("commerceShipmentId", commerceShipmentId);
			_params.put("description", checkNull(description));
			_params.put("expectedDateDay", expectedDateDay);
			_params.put("expectedDateHour", expectedDateHour);
			_params.put("expectedDateMinute", expectedDateMinute);
			_params.put("expectedDateMonth", expectedDateMonth);
			_params.put("expectedDateYear", expectedDateYear);
			_params.put("name", checkNull(name));
			_params.put("phoneNumber", checkNull(phoneNumber));
			_params.put("shippingDateDay", shippingDateDay);
			_params.put("shippingDateHour", shippingDateHour);
			_params.put("shippingDateMinute", shippingDateMinute);
			_params.put("shippingDateMonth", shippingDateMonth);
			_params.put("shippingDateYear", shippingDateYear);
			_params.put("status", status);
			_params.put("street1", checkNull(street1));
			_params.put("street2", checkNull(street2));
			_params.put("street3", checkNull(street3));
			_params.put("trackingNumber", checkNull(trackingNumber));
			_params.put("zip", checkNull(zip));

			_command.put("/commerce.commerceshipment/update-commerce-shipment", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateExpectedDate(long commerceShipmentId, int expectedDateMonth, int expectedDateDay, int expectedDateYear, int expectedDateHour, int expectedDateMinute) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceShipmentId", commerceShipmentId);
			_params.put("expectedDateDay", expectedDateDay);
			_params.put("expectedDateHour", expectedDateHour);
			_params.put("expectedDateMinute", expectedDateMinute);
			_params.put("expectedDateMonth", expectedDateMonth);
			_params.put("expectedDateYear", expectedDateYear);

			_command.put("/commerce.commerceshipment/update-expected-date", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateShippingDate(long commerceShipmentId, int shippingDateMonth, int shippingDateDay, int shippingDateYear, int shippingDateHour, int shippingDateMinute) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceShipmentId", commerceShipmentId);
			_params.put("shippingDateDay", shippingDateDay);
			_params.put("shippingDateHour", shippingDateHour);
			_params.put("shippingDateMinute", shippingDateMinute);
			_params.put("shippingDateMonth", shippingDateMonth);
			_params.put("shippingDateYear", shippingDateYear);

			_command.put("/commerce.commerceshipment/update-shipping-date", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateStatus(long commerceShipmentId, int status) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceShipmentId", commerceShipmentId);
			_params.put("status", status);

			_command.put("/commerce.commerceshipment/update-status", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

}